wordpress-mobile/WordPressKit-iOS
# Add ReaderSiteSearchServiceRemote and tests

This PR looks big, but most of it is mock data!

This PR adds a new remote: `ReaderSiteSearchServiceRemote`, which I'll use to fetch a list of possible Reader sites for a user to follow: https://github.com/wordpress-mobile/WordPress-iOS/issues/9433. The remote has a single method to perform a search, and returns the results.

Results should include a feed ID, a title, and a URL. They may also include an optional blavatar URL and description.

Equivalent Android PR: https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/822

**To test:**

* Check the code
* Build and run tests, and check everything passes
* Let me know if you'd like to actually try it out with WPiOS and I can push up a work in progress branch there that uses this remote.
